File: practice/Capstone_project/mini-ecommerce/backend/order-service/src/main/java/com/wipro/com/order_service/order/service/OrderService.java
package com.wipro.com.order_service.order.service;

import com.wipro.com.order_service.order.entity.Order;
import java.util.List;

public interface OrderService {
    
    // Create order from cart for a user
    Order createOrderFromCart(int userId);

    // Cancel an order
    Order cancelOrder(int orderId);

    // Update status of an order
    Order updateOrderStatus(int orderId, String status);

    // Get all orders
    List<Order> getAllOrders();

    // Get orders by user
    List<Order> getOrdersByUser(int userId);

    // Get order by orderId
    Order getOrderById(int orderId);
}

File: capstone_project/front_end/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import { API_BASE } from '../app.config';
import { HttpClient } from '@angular/common/http';
import { AuthService } from './auth.service';
import { Observable } from 'rxjs/internal/Observable';
import { Order } from '../models/order';
import { of } from 'rxjs/internal/observable/of';
import { CartItem } from '../models/cart';

@Injectable({
  providedIn: 'root'
})
export class OrderService {
 private base = API_BASE + '/orderms/order';
  constructor(private http: HttpClient, private auth: AuthService) {}

  // components call this to get current user's orders
  getMyOrders(): Observable<Order[]> {
  return this.http.get<Order[]>(`${this.base}/myorders`);
}


  placeOrderFromCart(items: CartItem[]): Observable<Order[]> {
    const payload = {
      items: items.map(i => ({ productId: i.productId, quantity: i.quantity }))
    };
    return this.http.post<Order[]>(`${this.base}/place`, payload);
  }
  getAllOrders(): Observable<Order[]> {
  return this.http.get<Order[]>(`${this.base}/admin/all`);
}

// admin: update status
updateStatus(orderId: number, status: string) {
  return this.http.put<Order>(`/api/orders/${orderId}/status`, { status });
}

cancelOrder(orderId: number) {
  return this.http.put<Order>(`/api/orders/${orderId}/cancel`, {});
}
updateOrderStatus(orderId: number, status: string) {
  // note: adapt this.base if your base is different; earlier you used API_BASE + '/orderms/order'
  return this.http.put<Order>(`${this.base}/${orderId}/status?status=${encodeURIComponent(status)}`, {});
}
}

